kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowActivityRunner.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.ui
+
+import android.app.Activity
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Parcelable
+import android.support.annotation.CheckResult
+import android.support.v4.app.FragmentActivity
+import com.squareup.workflow.Workflow
+import io.reactivex.Observable
+
+/**
+ * Packages a [Workflow] and a [ViewRegistry] to drive an [Activity][FragmentActivity].
+ *
+ * You'll never instantiate one of these yourself. Instead, use
+ * [FragmentActivity.setContentWorkflow]. See that method for more details.
+ */
+class WorkflowActivityRunner<OutputT : Any, RenderingT : Any>
+internal constructor(private val model: WorkflowViewModel<OutputT, RenderingT>) {
+
+  internal val renderings: Observable<out RenderingT> = model.updates.map { it.rendering }
+
+  val viewRegistry: ViewRegistry = model.viewRegistry
+
+  /**
+   * A stream of the [output][OutputT] values emitted by the [Workflow]
+   * managed by this model.
+   */
+  val output: Observable<out OutputT> = model.updates.filter { it.output != null }
```

Comment:
Nit:
`.filterNotNull().map { …`
or
`mapNonNull { it?.output }`

Reply:
I don't see either of those. Rx2 forbids nulls in the stream, no?

Reply:
Oh, those are internal, nevermind :(